core/node/api_server/src/web3/state.rs

```diff
@@ -241,6 +247,51 @@ impl SealedL2BlockNumber {
     }
 }
 
+#[derive(Debug, Clone)]
+pub(crate) struct BridgeAddressesHandle(Arc<RwLock<BridgeAddresses>>);
+
+impl BridgeAddressesHandle {
+    pub fn new(bridge_addresses: BridgeAddresses) -> Self {
+        Self(Arc::new(RwLock::new(bridge_addresses)))
+    }
+
+    /// Creates a handle to the bridge addresses together with a task that will update them on a schedule.
+    pub fn new_with_updater(
+        main_node_client: Box<DynClient<L2>>,
+        initial_value: BridgeAddresses,
+        update_interval: Duration,
+        stop_receiver: watch::Receiver<bool>,
+    ) -> (Self, impl Future<Output = anyhow::Result<()>>) {
+        let this = Self(Arc::new(RwLock::new(initial_value)));
+        let updater = this.clone();
+
+        let update_task = async move {
+            loop {
+                if *stop_receiver.borrow() {
+                    tracing::debug!("Stopping bridge addresses updates");
+                    return Ok(());
+                }
+
+                match main_node_client.get_bridge_contracts().await {
+                    Ok(bridge_addresses) => {
+                        *updater.0.write().await = bridge_addresses;
+                    }
+                    Err(err) => {
+                        tracing::error!("Failed to query `get_bridge_contracts`, error: {err:?}");
+                    }
+                }
+                tokio::time::sleep(update_interval).await;
```

Comment:
It makes sense to use [the `timeout` construction](https://github.com/matter-labs/zksync-era/blob/79b6fcf8b5d10a0ccdceb846370dd6870b6a32b5/core/node/api_server/src/web3/pubsub.rs#L74-L76), so that the node shutdown is more responsive. TBH, this is not done in the neighboring `SealedL2BlockNumber` task, but it's expected to have much lower sleep interval, at least judging by the default values. (Although ideally, it can be refactored as well.)